spyder/app/mainwindow.py

```diff
@@ -1087,7 +1018,7 @@ def pre_visible_setup(self):
                 if isinstance(plugin_instance.CUSTOM_LAYOUTS, list):
                     for custom_layout in plugin_instance.CUSTOM_LAYOUTS:
                         self.layouts.register_layout(
-                            self, custom_layout)
+                            self.layouts, custom_layout)
```

Comment:
Shouldn't this be `plugin_instance` instead of `self.layouts`? Also, maybe moving this code to  register custom layouts inside the Layout plugin and instead here call something like `self.layouts.register_custom_layouts(PLUGIN_REGISTRY.external_plugins)` could be worthy?

Reply:
> Shouldn't this be plugin_instance instead of self.layouts?

Nop, it needs to be the Layout plugin instance.

> Also, maybe moving this code to register custom layouts inside the Layout plugin

Good idea. I'll implement it right away.